test/unit/mama/ng/scheduler/ScheduleSpec.groovy

```diff
@@ -0,0 +1,665 @@
+package mama.ng.scheduler
+
+import grails.test.mixin.Mock
+import grails.test.mixin.TestFor
+import grails.test.mixin.TestMixin
+import grails.test.mixin.domain.DomainClassUnitTestMixin
+import groovy.time.TimeCategory
+import org.codehaus.groovy.runtime.typehandling.GroovyCastException
+import spock.lang.Specification
+
+/**
+ * See the API for {@link grails.test.mixin.domain.DomainClassUnitTestMixin} for usage instructions
+ */
+@TestMixin(DomainClassUnitTestMixin)
+@TestFor(Schedule)
+@Mock([Schedule])
+class ScheduleSpec extends Specification {
+
+    Schedule schedule
+
+    def setup() {
+        use (TimeCategory) {
+            schedule = new Schedule(
+                userId: "user-id",
+                frequency: 5,
+                cronDefinition: "0 9 * * 3",
+                nextSend: new Date() + 2.days,
+                endpoint: "http://test.endpoint.com"
+            )
+        }
+    }
+
+    def cleanup() {
+        schedule.delete()
+    }
+
+    void "test everything works"() {
+        use (TimeCategory) {
+            schedule.userId = "user-id"
+            schedule.frequency = 5
+            schedule.cronDefinition = "0 9 * * 3"
+            schedule.nextSend = new Date() + 2.days
+            schedule.endpoint = "http://test.endpoint.com"
+        }
+        schedule.save()
+        expect:
+        !schedule.hasErrors()
+    }
+
+    void "test blank user id"() {
+        given:
+        schedule.userId = ""
+        schedule.save()
```

Comment:
Should these be indented more? I'm mostly coding in Python so I'm sensitive to indents. :)